src/main/java/uk/gov/hmcts/darts/cache/token/ServiceContextCacheValue.java

```diff
@@ -1,20 +1,131 @@
 package uk.gov.hmcts.darts.cache.token;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import documentum.contextreg.BasicIdentity;
+import documentum.contextreg.Identity;
+import documentum.contextreg.ObjectFactory;
 import documentum.contextreg.ServiceContext;
-import lombok.RequiredArgsConstructor;
+import jakarta.xml.bind.JAXBContext;
+import jakarta.xml.bind.JAXBElement;
+import jakarta.xml.bind.JAXBException;
+import  jakarta.xml.bind.Marshaller;
+import jakarta.xml.bind.Unmarshaller;
+import org.springframework.xml.transform.StringSource;
+import uk.gov.hmcts.darts.cache.token.exception.CacheException;
 
-@RequiredArgsConstructor
+import java.io.StringWriter;
+
+@JsonTypeName("ServiceContextCacheValue")
 public class ServiceContextCacheValue implements RefreshableCacheValue {
+    private String contextStr;
+
+    private String id;
+
+    private ServiceContext context;
+
+    private String token;
+
+    protected static final String EMPTY_DOWN_STREAM_TOKEN = "";
+
+    public ServiceContextCacheValue() throws CacheException {
+
+    }
+
+    public ServiceContextCacheValue(ServiceContext context) throws CacheException {
+
+        try {
+            JAXBElement<ServiceContext> scontext =  new ObjectFactory().createServiceContext(context);
+            StringWriter sw = new StringWriter();
+            JAXBContext jaxbContext = JAXBContext.newInstance(ServiceContext.class);
+            Marshaller jaxbMarshaller = jaxbContext.createMarshaller();
+            jaxbMarshaller.marshal(scontext, sw);
+            contextStr = sw.toString();
+        } catch (JAXBException e) {
+            throw new CacheException(e);
+        }
+    }
+
+    public ServiceContextCacheValue(ServiceContextCacheValue value) {
+        setId(value.getId());
+        setContextString(value.getContextString());
+        setDownstreamToken(value.getDownstreamToken());
+    }
+
+    public String getContextString() {
+        return contextStr;
+    }
+
+    public void setContextString(String context) {
+        this.contextStr = context;
+    }
+
+    @JsonIgnore
+    public ServiceContext getServiceContext() throws CacheException {
+
+        if (context == null) {
+            try {
+                JAXBContext jaxbContext = JAXBContext.newInstance(ServiceContext.class);
```

Comment:
I think this JAXBContext.newInstance stuff should only be done in the constructor, so we would use a class field like in SoapHeaderConverter?

Reply:
yep you are right. Done